mlir/lib/Dialect/Affine/IR/AffineOps.cpp

```diff
@@ -2562,7 +2562,7 @@ bool mlir::affine::isAffineInductionVar(Value val) {
 
 AffineForOp mlir::affine::getForInductionVarOwner(Value val) {
   auto ivArg = llvm::dyn_cast<BlockArgument>(val);
-  if (!ivArg || !ivArg.getOwner())
+  if (!ivArg || !ivArg.getOwner() || !ivArg.getOwner()->getParent())
     return AffineForOp();
   auto *containingInst = ivArg.getOwner()->getParent()->getParentOp();
   if (auto forOp = dyn_cast<AffineForOp>(containingInst))
```

Comment:
```suggestion
  if (!ivArg || !ivArg.getOwner())
    return AffineForOp();
  if (auto forOp = ivArg.getOwner()->getParentOfType<AffineForOp()))
```

This seems like the most straightforward fix to me.

Reply:
Thanks @joker-eph for the suggestion. But "`ivArg.getOwner()`" will be a "`Block*`", and the "`getParentOfType()`" method is not available on that. So we need to access to the "`Region*`" to get that method. 
I have modified my code.

Reply:
> But "ivArg.getOwner()" will be a "Block*"

Oh OK, reading the code here I thought it would be an `Operation *`.